src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStoragePublisher.java

```diff
@@ -435,9 +435,15 @@ public synchronized void perform(
                     zipArchiveBlob = serviceData.getArchiveBlobs().get(0);
                 }
 
-                run.addAction(new AzureBlobAction(expContainerName, expShareName, storageType,
-                        serviceData.getIndividualBlobs(), zipArchiveBlob, allowAnonymousAccess,
-                        getStorageCredentialId()));
+                List<AzureBlob> individualBlobs = serviceData.getIndividualBlobs();
+                AzureBlobAction existAction = run.getAction(AzureBlobAction.class);
+                if (existAction != null) {
+                    existAction.getIndividualBlobs().addAll(individualBlobs);
+                } else {
+                    run.addAction(new AzureBlobAction(expContainerName, expShareName, storageType,
```

Comment:
is this scenario testable? how about add a test?

Reply:
It will be a little complicate to write test codes here, and an action refers to a page resource in Jenkins. The change here will only affect the page view. I have verified it works properly now.